modules/tensor_mechanics/include/kernels/CrackPropagationHeatEnergy.h

```diff
@@ -0,0 +1,66 @@
+/****************************************************************/
```

Comment:
It seems like this might be better named `CrackPropagationHeatSource` (for consistency with the `HeatSource` kernel we have in the heat_conduction module.

Also, I'm torn about where we should put this -- it involves three modules: tensor_mechanics, heat_conduction, and phase_field. Since it is assembling to the residual of the temperature variable, I think the best place for it is in the heat_conduction module.